crypto/dh/dh_key.c

```diff
@@ -358,8 +371,21 @@ static int generate_key(DH *dh)
     if (!ossl_dh_generate_public_key(ctx, dh, priv_key, pub_key))
         goto err;
 
+#ifdef FIPS_MODULE
+    if (DH_check_pub_key(dh, pub_key, &validate) <= 0) {
```

Comment:
Key checks should not be necessary if the keygen is approved. (i.e. SP800-56A keygen).

Reply:
Noted, thanks!